### ISSUE TITLE
Support a ":closest" procedural cosmetic filter

### ISSUE THREAD

**Issue by Tomalak** [uBlockOrigin/uBlock-issues#2190]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

Akin to jQuery's `.closest()` API, it would be convenient to be able to select the first matching element at a non-fixed point upward in the DOM tree via CSS. Currently there are only e.g. `:upward(2)` and `:xpath(../..)`, both of which select fixed points.

The best approximation for jQuery's `.closest('.someClass')` would be a rather unwieldy `:xpath(./ancestor::*[contains(concat(' ', @class, ' '), concat(' ', 'someClass', ' '))][1])`, with the spaces thrown in to prevent partial matches. The equivalent XPath grows considerably more complex with every additional CSS selector.

A friendlier and more versatile alternative could be `:closest(.someClass)`, with `.someClass` being _"any XPath expression"_.

### A specific URL where the issue occurs

n/a

### Steps to Reproduce

n/a

### Expected behavior

n/a

### Actual behavior

n/a

### uBlock Origin version

n/a

### Browser name and version

n/a

### Operating System and version

n/a

**Comment by gorhill** [comment]
Just use `:upward(.someClass)`. [`:upward()` documentation](https://github.com/gorhill/uBlock/wiki/Procedural-cosmetic-filters#subjectupwardarg):

>  arg:
> 
> - Positive number >= 1 and < 256, distance from the currently selected node.
> - A valid plain CSS selector.

**Comment by Tomalak** [comment]
Whoops, I missed that part of the documentation. Awesome! Thank you.

**Comment by gorhill** [comment]
Note that when you request a feature, you need to provide real worlds cases of why these features are needed. So even if `:upward()` did not do what you are actually asking, I would have declined: there has to be good reasons provided to support a feature requests, and they have to be related to actual, real world cases (personal requests or contrived cases do not qualify).

**Comment by Tomalak** [comment]
@gorhill Well, the use case would have been that it allows writing filters that are more resilient against structural changes in a website, things like "another <div> wrapper gets added to achieve some visual effect".

But it already exists, and I just did not see it in the docs because of my sloppy reading, sorry for that. I went from `:nth-ancestor(n)` and kind of expected `:upward()` just to be a rename that behaves the same.
